build-tools/automation/yaml-templates/apk-instrumentation.yaml

```
@@ -6,31 +6,49 @@ parameters:
   extraBuildArgs: ""
   testResultsFormat: NUnit
   packageType: Apk
-  artifactName: ""
+  artifactSource: ""
   artifactFolder: ""
+  useDotNetPreview: false
   condition: succeeded()
 
 steps:
-- task: MSBuild@1
-  displayName: run ${{ parameters.testName }}
-  inputs:
-    solution: ${{ parameters.project }}
-    configuration: ${{ parameters.configuration }}
-    msbuildArguments: >-
-      /restore
-      /t:AcquireAndroidTarget,SignAndroidPackage,DeployTest${{ parameters.packageType }}s,CheckAndRecordApkSizes,RunTestApks,UndeployTestApks,RenameApkTestCases,ReportComponentFailures
-      /bl:$(System.DefaultWorkingDirectory)/bin/Test${{ parameters.configuration }}/run-${{ parameters.testName }}.binlog
-      ${{ parameters.extraBuildArgs }}
-  condition: ${{ parameters.condition }}
-  continueOnError: true
+- ${{ if ne(parameters.useDotNetPreview, true) }}:
+  - task: MSBuild@1
+    displayName: run ${{ parameters.testName }}
+    inputs:
+      solution: ${{ parameters.project }}
+      configuration: ${{ parameters.configuration }}
+      msbuildArguments: >-
+        /restore
+        /t:AcquireAndroidTarget,SignAndroidPackage,DeployTest${{ parameters.packageType }}s,CheckAndRecordApkSizes,RunTestApks,UndeployTestApks,RenameApkTestCases,ReportComponentFailures
+        /bl:$(System.DefaultWorkingDirectory)/bin/Test${{ parameters.configuration }}/run-${{ parameters.testName }}.binlog
+        ${{ parameters.extraBuildArgs }}
+    condition: ${{ parameters.condition }}
+    continueOnError: true
+
+- ${{ if eq(parameters.useDotNetPreview, true) }}:
+  - powershell: |
+      $dotnetPath = "$env:HOME/Library/Android/dotnet/dotnet"
+      if ([Environment]::OSVersion.Platform -eq "Unix") {
```

Could probably make it a little clearer to swap these two lines around:

```suggestion
      if ([Environment]::OSVersion.Platform -eq "Unix") {
          $dotnetPath = "$env:HOME/Library/Android/dotnet/dotnet"
```